lib/core/test/unit/Cardano/Wallet/Primitive/TypesSpec.hs

```diff
@@ -659,6 +667,11 @@ makeRangeValid = \case
     Range (Just p) (Just q) -> Range (Just $ min p q) (Just $ max p q)
     r -> r
 
+rangeIsSubrangeOf :: Ord a => Range a -> Range a -> Bool
+rangeIsSubrangeOf (Range a b) r =
+    maybe True (`isWithinRange` r) a
```

Comment:
The `True` case is wrong though I think 🙃

Reply:
I think you're right to be suspicious!

For example, the following expression evaluates to `True`:

```hs
Range Nothing Nothing `rangeIsSubrangeOf` Range (Just 0) (Just 0)
```

I've made a PR here with a property-tested implementation of `isSubrangeOf`:

https://github.com/input-output-hk/cardano-wallet/pull/653